### PR TITLE
Fix broken links

### LINKED ISSUE #3568: Fix broken links
**From @nak3** comment on https://github.com/knative/docs/issues/3543

As README.md was changed to index.md, we need to remove README.md.

Not only samples link, but also all README.md link grep-ed by:

$ grep -rH "\/README.md)" | grep -v https

need to be updated.

cc @evankanderson

### PR DESCRIPTION
**Pull request by NovaHe**
Fixes #3568 #3577

I use the following command to scan in multiple threads
```shell
find . -type f -print0  | xargs -0 -P 6 grep -rni "/README.md)"|grep -v https
```

```shell
./docs/serving/samples/rest-api-go/index.md:15:1. A Kubernetes cluster with [Knative Serving](../../../install/README.md) installed
./docs/serving/samples/rest-api-go/index.md:205:The [traffic splitting example](../traffic-splitting/README.md) continues from
./docs/serving/samples/knative-routing-go/index.md:23:1. A Kubernetes cluster with [Knative Serving](../../../install/README.md)
./docs/serving/samples/multi-container/index.md:13:  [installation instructions](../../../install/README.md) if you need to
./docs/serving/samples/hello-world/helloworld-java-spring/README.md:17:  [installation instructions](../../../../install/README.md) if you need to
./docs/serving/samples/grpc-ping-go/index.md:19:- [Install the latest version of Knative Serving](../../../install/README.md).
./docs/serving/samples/blue-green-deployment.md:17:- A Kubernetes cluster with [Knative installed](../../install/README.md).
./docs/serving/samples/secrets-go/index.md:25:  [installation instructions](../../../install/README.md) if you need to create
./docs/serving/samples/gitwebhook-go/index.md:19:    cluster, see the [installation guides](../../../install/README.md).
./docs/eventing/accessing-traces.md:14:You must have a Knative cluster running with the Eventing component installed. [Learn more](../install/README.md)
./docs/eventing/debugging/index.md:14:[Knative Eventing](../README.md). You don't need to be an expert, but do need to
./docs/eventing/debugging/index.md:19:1. Setup [Knative Eventing and an Eventing-contrib resource](../README.md).
./docs/eventing/samples/helloworld/helloworld-python/_index.md:11:However, you can also deploy the app as a [Knative Serving Service](../../../../serving/README.md).
./docs/eventing/samples/helloworld/helloworld-go/_index.md:19:[Knative Serving Service](../../../../serving/README.md).
./docs/eventing/samples/helloworld/helloworld-go/_index.md:283:We can send an http request directly to the [Broker](../../../broker/README.md)
./docs/eventing/samples/helloworld/_index.md:16:However, you can also deploy the app as a [Knative Serving Service](../../../serving/README.md)
./docs/eventing/samples/helloworld/_index.md:23:    [Knative Serving](../../../install/README.md).
./docs/eventing/samples/parallel/multiple-branches/_index.md:17:Please refer to the sample overview for the [prerequisites](../README.md).
./docs/eventing/samples/parallel/mutual-exclusivity/_index.md:12:[multiple branches example](../multiple-branches/README.md) except that we are
./docs/eventing/samples/parallel/mutual-exclusivity/_index.md:21:Please refer to the sample overview for the [prerequisites](../README.md).
./docs/eventing/samples/parallel/_index.md:25:[`PingSource`](../ping-source/README.md) as the source of events.
./docs/eventing/samples/parallel/_index.md:33:- [Parallel with multiple branches and global reply](./multiple-branches/README.md)
./docs/eventing/samples/parallel/_index.md:34:- [Parallel with mutually exclusive cases](./mutual-exclusivity/README.md)
./docs/eventing/samples/kafka/source/index.md:8:Tutorial on how to build and deploy a `KafkaSource` [Eventing source](../../../sources/README.md) using a Knative Serving `Service`.
./docs/eventing/samples/kafka/source/index.md:12:- Ensure that you meet the [prerequisites listed in the Apache Kafka overview](../README.md).
./docs/eventing/samples/kafka/source/index.md:13:- A Kubernetes cluster with [Knative Kafka Source installed](../../../../install/README.md).
./docs/eventing/samples/kafka/binding/index.md:21:   [prerequisites listed in the Apache Kafka overview](../README.md).
./docs/eventing/samples/kafka/channel/index.md:14:  [prerequisites listed in the Apache Kafka overview](../README.md).
./docs/eventing/samples/kafka/channel/index.md:16:  [Knative Kafka Channel installed](../../../../install/README.md).
./docs/eventing/samples/kafka/_index.md:100:- [`KafkaSource` to `Service`](./source/README.md)
./docs/eventing/samples/kafka/_index.md:101:- [`KafkaChannel` and Broker](./channel/README.md)
./docs/eventing/samples/kafka/_index.md:102:- [`KafkaBinding`](./binding/README.md)
./docs/eventing/samples/gitlab-source/index.md:18:   Follow the [installation instructions](../../../install/README.md) if you
./docs/eventing/samples/_index.md:11:[Learn more about Knative Eventing and Eventing Sources](../README.md).
./docs/eventing/flows/parallel.md:47:Learn how to use Parallel by following the [examples](../samples/parallel/README.md)
./vendor/knative.dev/hack/README.md:114:   [here](../kntest/pkg/kubetest2/gke/README.md) to the `initialize` function

```
## Proposed Changes 

- Fix broken links

### CODE REVIEW

**Comment by NovaHe** [comment]
cc @abrennan89 PTAL, thanks

File: docs/eventing/accessing-traces.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ visualize and trace your requests.
 
 ## Before you begin
 
-You must have a Knative cluster running with the Eventing component installed. [Learn more](../install/README.md)
+You must have a Knative cluster running with the Eventing component installed. [Learn more](../install/_index.md)
```
Comment (abrennan89) [inline docs/eventing/accessing-traces.md:14]:
```suggestion
You must have a Knative cluster running with the Eventing component installed. [Learn more](../install/)
```

There's no need to include `_index.md` for any of these, adding the path to the directory is enough AFAIK

Comment (NovaHe) [reply docs/eventing/accessing-traces.md]: fixed

File: docs/eventing/channels/channel-types-defaults.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ Knative provides the InMemoryChannel channel implementation by default. This def
 **NOTE:** InMemoryChannel channels should not be used in production environments.
 
 The default channel implementation is specified in the `default-ch-webhook` ConfigMap in the `knative-eventing` namespace.
-For more information about modifying ConfigMaps, see [Configuring the Eventing Operator custom resource](./docs/install/configuring-eventing-cr/).
+For more information about modifying ConfigMaps, see [Configuring the Eventing Operator custom resource](../../../docs/install/operator/configuring-eventing-cr.md).
```
Comment (abrennan89) [inline docs/eventing/channels/channel-types-defaults.md:13]:
```suggestion
For more information about modifying ConfigMaps, see [Configuring the Eventing Operator custom resource](../../../docs/install/operator/configuring-eventing-cr).
```

**Comment by abrennan89** [comment]
@NovaHe have you tested the links using a local build of the docs?

File: docs/eventing/channels/channel-types-defaults.md (hunk 2 above)
Comment (NovaHe) [reply docs/eventing/channels/channel-types-defaults.md]: done! thanks

**Comment by abrennan89** [comment]
No problem at all @NovaHe - thanks so much for the preview link!
/lgtm
/approve

**Pull request merged**